src/Domain/CoordinateMaps/UniformCylindricalEndcap.cpp

```diff
@@ -575,84 +576,122 @@ std::optional<std::array<double, 3>> UniformCylindricalEndcap::inverse(
       // Figure that out by looking at the deriv of the function.
       const double deriv_function_at_rhobar_min =
           function_and_deriv_to_zero(rhobar_min).second;
-      // The formula below would be Newton-Raphson except for the
-      // factor of 2. The factor of 2 is there to over-compensate so that
-      // the new rhobar_min brackets the root.
-      const double new_rhobar_min =
-          rhobar_min -
-          2.0 * function_at_rhobar_min / deriv_function_at_rhobar_min;
-      try {
-        // We have a good guess, so use newton_raphson
-        const size_t digits = 14;
-        // use rhobar_min as maximum value, since that brackets the root.
-        rhobar =
-            RootFinder::newton_raphson(function_and_deriv_to_zero, rhobar_min,
-                                       new_rhobar_min, rhobar_min, digits);
-      } catch (std::exception&) {
-        const double function_at_new_rhobar_min =
-            function_to_zero(new_rhobar_min);
+      const double trial_rhobar_min_increment =
+          -2.0 * function_at_rhobar_min / deriv_function_at_rhobar_min;
+      // new_rhobar_min = rhobar_min + trial_rhobar_min_increment would be
+      // a Newton-Raphson step except for the factor of 2 in
+      // trial_rhobar_min_increment. The factor of 2 is there to
+      // over-compensate so that the new rhobar_min brackets the
+      // root.
+      //
+      // But sometimes the factor of 2 is not enough when
+      // trial_rhobar_min_increment is roundoff-small so that the
+      // change in new_rhobar_min is zero or only in the last one or
+      // two bits.  If this is the case, then we set
+      // rhobar_min_increment to some small roundoff value with the
+      // same sign as trial_rhobar_min_increment.
+      // Note that rhobar is always between zero and one, so it is
+      // ok to use an absolute epsilon.
+      const double rhobar_min_increment =
+          abs(trial_rhobar_min_increment) >
+                  100.0 * std::numeric_limits<double>::epsilon()
+              ? trial_rhobar_min_increment
+              : boost::math::copysign(
```

Comment:
https://en.cppreference.com/w/cpp/numeric/math/copysign ?
We also have `sgn` in `Utilities/Math.hpp` depending on exactly what you want